src/update.ts

```diff
@@ -268,14 +268,18 @@ export class Updater {
     return current === updated
   }
 
-  private async determineChannel(): Promise<string> {
-    const channelPath = path.join(this.config.dataDir, 'channel')
-    if (fs.existsSync(channelPath)) {
-      const channel = await fs.readFile(channelPath, 'utf8')
-      return String(channel).trim()
+  private async determineChannel(version:string|undefined): Promise<string> {
+    try {
+      const {body} = await HTTP.get(`https://registry.npmjs.org/${this.config.pjson.name}/`)
+      const tags = (body as {'dist-tags': Record<string, string>})['dist-tags']
+      const tag = Object.keys(tags).find(v => tags[v] === version) ?? 'stable'
+      // convert from npm style tag defaults to OCLIF style
+      if (tag === 'latest') return 'stable'
+      if (tag === 'latest-rc') return 'stable-rc'
+      return tag
+    } catch {
+      return 'stable'
```

Comment:
is `stable` the right default?  Would it be more accurate to fall back to what's in the channel file, like the original implementation did?